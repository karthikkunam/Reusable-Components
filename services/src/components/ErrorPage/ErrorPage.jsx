import React from 'react';
import PropTypes from 'prop-types';

function ErrorPage(props) {
    const { header, content } = props;
    return (
        <div className="row component-example mb-3 full-height">
            <div className="col-sm-12">
                <div className="error-page full-height">
                    <div className="row full-height col">
                        <div className="col-sm-8 col-left">
                            <div className="center-align error-page-message">
                                <h1 className="header">{header}</h1>
                                <p className="content">{content}</p>
                            </div>
                        </div>
                        <div className="col-sm-4 col-right">
                            <span className="icon icon-logo-7eleven-color center-align"><span className="path1"></span><span className="path2"></span><span className="path3"></span><span className="path4"></span><span className="path5"></span><span className="path6"></span><span className="path7"></span><span className="path8"></span><span className="path9"></span><span className="path10"></span></span>
                        </div>
                        <div className="footer">
                            <div className="bar"></div>
                            <div className="bar"></div>
                            <div className="bar"></div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
};

ErrorPage.propTypes = {
    header: PropTypes.string,
    content: PropTypes.string
};

ErrorPage.defaultProps ={
    header: 'Session Expired',
    content: 'Your 7BOSS session has expired. To reconnect, please reboot your browser and login again.'
};

export default ErrorPage;
