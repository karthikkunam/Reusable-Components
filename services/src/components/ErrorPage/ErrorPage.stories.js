import React from 'react';
import ErrorPage from '.';

export default {
    title: 'ErrorPage',
    component: ErrorPage
};

export const ErrorFullPage = () => (
    <>
    <ErrorPage 
    header="Session Expired"
    content="Your 7BOSS session has expired. To reconnect, please reboot your browser and login again."
    />
    </>
);

