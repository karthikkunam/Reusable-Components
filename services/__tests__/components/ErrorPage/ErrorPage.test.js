import React from 'react';
import { shallow } from 'enzyme';
import ErrorPage from '../../../src/components/ErrorPage';

describe('ErrorPage Component', () => {
  it('renders correctly', () => {
    const component = shallow(<ErrorPage/>);
    expect(component).toMatchSnapshot();
  });
});