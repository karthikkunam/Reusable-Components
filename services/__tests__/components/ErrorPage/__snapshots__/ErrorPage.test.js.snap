// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage Component renders correctly 1`] = `
<div
  className="row component-example mb-3 full-height"
>
  <div
    className="col-sm-12"
  >
    <div
      className="error-page full-height"
    >
      <div
        className="row full-height col"
      >
        <div
          className="col-sm-8 col-left"
        >
          <div
            className="center-align error-page-message"
          >
            <h1
              className="header"
            >
              Session Expired
            </h1>
            <p
              className="content"
            >
              Your 7BOSS session has expired. To reconnect, please reboot your browser and login again.
            </p>
          </div>
        </div>
        <div
          className="col-sm-4 col-right"
        >
          <span
            className="icon icon-logo-7eleven-color center-align"
          >
            <span
              className="path1"
            />
            <span
              className="path2"
            />
            <span
              className="path3"
            />
            <span
              className="path4"
            />
            <span
              className="path5"
            />
            <span
              className="path6"
            />
            <span
              className="path7"
            />
            <span
              className="path8"
            />
            <span
              className="path9"
            />
            <span
              className="path10"
            />
          </span>
        </div>
        <div
          className="footer"
        >
          <div
            className="bar"
          />
          <div
            className="bar"
          />
          <div
            className="bar"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
